tendrl/node_agent/monitoring/collectd/collectors/gluster/heavy_weight/tendrl_glusterfs_profile_info.py

```diff
@@ -70,150 +68,148 @@
+                            'cumulativeStats/totalRead'
+                        ).text,
+                        'totalWrite': brick.find(
+                            'cumulativeStats/totalWrite'
+                        ).text
+                    },
+                    'intervalStats': {
+                        'blockStats': blkInterval,
+                        'fopStats': fopInterval,
+                        'duration': brick.find('intervalStats/duration').text,
+                        'totalRead': brick.find(
+                            'intervalStats/totalRead'
+                        ).text,
+                        'totalWrite': brick.find(
+                            'intervalStats/totalWrite'
+                        ).text
+                    }
                 }
             )
-        bricks.append(
-            {
-                brickKey: brickName,
-                'cumulativeStats': {
-                    'blockStats': blkCumulative,
-                    'fopStats': fopCumulative,
-                    'duration': brick.find(
-                        'cumulativeStats/duration'
-                    ).text,
-                    'totalRead': brick.find(
-                        'cumulativeStats/totalRead'
-                    ).text,
-                    'totalWrite': brick.find(
-                        'cumulativeStats/totalWrite'
-                    ).text
-                },
-                'intervalStats': {
-                    'blockStats': blkInterval,
-                    'fopStats': fopInterval,
-                    'duration': brick.find('intervalStats/duration').text,
-                    'totalRead': brick.find(
-                        'intervalStats/totalRead'
-                    ).text,
-                    'totalWrite': brick.find(
-                        'intervalStats/totalWrite'
-                    ).text
-                }
-            }
-        )
-    status = {
-        'volumeName': tree.find("volProfile/volname").text,
-        bricksKey: bricks
-    }
-    return status
+        status = {
+            'volumeName': tree.find("volProfile/volname").text,
+            bricksKey: bricks
+        }
+        return status
 
-
-def get_volume_profile_info(volName, cluster_id):
-    ret_val = {}
-    brickName = ''
-    profile_info = {}
-    for trial_cnt in xrange(0, 3):
-        profile_cmd_op, profile_err = tendrl_glusterfs_utils.exec_command(
-            "gluster volume profile %s info --xml" % volName
-        )
-        if profile_err:
-            if trial_cnt == 2:
-                collectd.error(
-                    'Failed to fetch profile info. The error is: %s' % (
-                        profile_err
+    def get_volume_profile_info(self, volName, cluster_id):
+        ret_val = {}
+        brickName = ''
+        profile_info = {}
+        for trial_cnt in xrange(0, 3):
+            profile_cmd_op, profile_err = tendrl_glusterfs_utils.exec_command(
```

Comment:
Shouldnt there be a sleep between next tries?

Reply:
Aah Yes I missed this